app/javascript/controllers/progress_controller.js

```diff
@@ -21,7 +21,18 @@ export default class extends Controller {
     if (stepName === 'file') {
       isComplete = document.querySelectorAll('[type=hidden][name="work[files][]"]').length > 0
     }
-
     step.classList.toggle('active', isComplete)
   }
+
+  displayErrors(event) {
```

Comment:
Here's the JS glue needed to render the server-generated errors on the client side.